pkg/controller/reconciler/flux_controller.go

```diff
@@ -0,0 +1,431 @@
+package reconciler
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/radius-project/radius/pkg/cli/bicep"
+	"github.com/radius-project/radius/pkg/cli/filesystem"
+	sdkclients "github.com/radius-project/radius/pkg/sdk/clients"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+
+	radappiov1alpha3 "github.com/radius-project/radius/pkg/controller/api/radapp.io/v1alpha3"
+)
+
+const (
+	deploymentTemplateRepositoryField = "spec.repository"
+	radiusConfigFileName              = "radius-gitops-config.yaml"
+	armJSONParametersKeyName          = "parameters"
+)
+
+// FluxController watches GitRepository objects for revision changes
+// and processes the artifacts fetched from the Source Controller.
+// It reads the git repository configuration, builds the bicep files.
+// specified in the configuration, and creates DeploymentTemplate objects
+// on the cluster.
+type FluxController struct {
+	client.Client
+	Bicep          bicep.Interface
+	FileSystem     filesystem.FileSystem
+	ArchiveFetcher ArchiveFetcher
+}
+
+// RadiusGitOpsConfig is the configuration for Radius in a Git repository.
+type RadiusGitOpsConfig struct {
+	Config []BicepConfig `yaml:"config"`
+}
+
+// BicepConfig is the build configuration for a Bicep file in a Git repository.
+type BicepConfig struct {
```

Comment:
Could we use a different name for this? `bicepconfig` in Bicep and the `BicepConfig` definition here have 2 different meanings. I think we should keep parity with Bicep as much as possible

Reply:
true. changing to `ConfigEntry`